Practice

acsolle66 opened the pull request.

Commit:
practicing abstract methods

Commit:
Redesigned codebase

Commit:
Refractored the file system in to smaller packages, implemented new class UserInterface

Samyssmile commented on Main.java:
```diff
@@ -0,0 +1,17 @@
+    DecryptionKeyGenerator demap = new DecryptionKeyGenerator();
+
+    enmap.generateKeyMap(Integer.valueOf(args[0]));
+    System.out.println(enmap.getCipherKey());
```
Use Logger instead

Samyssmile commented on RotCipher.java:
```diff
@@ -61,4 +50,4 @@ public static void main(String[] args) {
 		System.out.println("Press any key to EXIT");
 		String exit = scanner.nextLine();
```
No need to assign to exit variable. You dont use reused it

Samyssmile commented on rot/cipher/Cipher.java:
```diff
@@ -2,6 +2,17 @@
+	characterMap = new CharacterMap(rot);
+	}
+
+	abstract public String translateText() {
```
Remove {} brackets. public abstract String translateText(); An abstract methon dont have body

Samyssmile commented on rot/cipher/Cipher.java:
```diff
@@ -40,4 +51,20 @@ public String cipherText(String text) {
+public class Encrypt extends Cipher{
+	@Override
+	public String translateText() {
+		char[]
```
????

Samyssmile commented on rot/cipher/Cipher.java:
```diff
@@ -40,4 +51,20 @@ public String cipherText(String text) {
+public class Decrypt extends Cipher{
+	@Override
+	public String translateText() {
+
```
?

Samyssmile commented on rot/cipher/DecryptionKeyGenerator.java:
```diff
@@ -0,0 +1,21 @@
+public class DecryptionKeyGenerator extends CipherKeyGenerator{
+    @Override
+    public void generateKeyMap(int shift) {
+        EnglishAlphabetGenerator ABCGenerator = new EnglishAlphabetGenerator();
```
Suggestion
public class DecryptionKeyGenerator extends CipherKeyGenerator {

    @Override
    public void generateKeyMap(int shift) {
        char[] alphabet = new EnglishAlphabetGenerator().getABC();
        int alphabetLength = alphabet.length;
        int adjustedShift = super.adjustShiftToCharsetLength(shift, alphabetLength);

        createCipherMapping(alphabet, adjustedShift);
    }

    private void createCipherMapping(char[] alphabet, int shift) {
        int alphabetLength = alphabet.length;
        for (int i = 0; i < alphabetLength; i++) {
            char keyChar = alphabet[i];
            char valueChar = alphabet[(i + shift) % alphabetLength];
            CipherKeyMap.put(valueChar, keyChar);
        }
    }
}

Samyssmile commented on algorithms/Decrypt.java:
```diff
@@ -0,0 +1,12 @@
+
+public class Decrypt extends Cipher {
+  public Decrypt(int shift) {
+    super();
```
remove this super() call.

Samyssmile commented on algorithms/Encrypt.java:
```diff
@@ -0,0 +1,12 @@
+
+public class Encrypt extends Cipher {
+  public Encrypt(int shift) {
+    super();
```
Dont need this super call

Samyssmile commented on keys/CipherKeyGenerator.java:
```diff
@@ -0,0 +1,26 @@
+    return adjustedShift;
+  }
+
+  public HashMap<Character, Character> getKey() {
```
Check the Polyphormism. Im sure you need public Map<... here not HashMap
https://www.javatpoint.com/runtime-polymorphism-in-java

Commit:
removed super() from Encrypt and Decrypt files

Commit:
Simplified cipher key generation
Replaced the adjustShiftToCharsetLength with a simpler method.
Saved for further work...

Samyssmile commented on algorithms/Cipher.java:
```diff
@@ -3,10 +3,11 @@
 public abstract class Cipher {
   protected char[] charSet;
-  protected HashMap<Character, Character> characterMap;
+  protected Map<Character, Character> characterMap;
```
good
